dart/lib/src/sentry_client.dart

```diff
@@ -106,8 +106,13 @@ class SentryClient {
         return _sentryId;
       }
     }
-    final envelope = SentryEnvelope.fromEvent(preparedEvent, _options.sdk);
-    return await _options.transport.send(envelope);
+    final envelope = SentryEnvelope.fromEvent(
+      preparedEvent,
+      _options.sdk,
+      attachments: scope?.attachements,
```

Comment:
@philipphofmann does attachment require to be in the same envelope as the event?

otherwise https://github.com/getsentry/sentry-java/issues/1508 will be a blocker

Reply:
yep, ideally, it'd be within the same envelope, but Android opens up and break it down because it needs to enrich the event before sending.

Reply:
> @philipphofmann does attachment require to be in the same envelope as the event?

No, they don't, but should, see https://develop.sentry.dev/sdk/envelopes/#attachment

> For minidump and apple crash report attachments, the corresponding "event" Item must be sent within the same Envelope.
> Generic attachments can be ingested separately from their events. We recommend sending them in the same Envelope, which allows for more efficient rate limiting and filtering.

Reply:
yep, I know about minidumps, so at least https://github.com/getsentry/sentry-java/issues/1508 isn't a blocker for now